adafruit_usb_host_descriptors.py

```diff
@@ -83,32 +101,123 @@ def get_configuration_descriptor(device, index):
     return full_buf
 
 
-def _find_boot_endpoint(device, protocol_type: Literal[PROTOCOL_MOUSE, PROTOCOL_KEYBOARD]):
+def get_report_descriptor(device, interface_num, length):
+    """
+    Fetches the HID Report Descriptor.
+    This tells us what the device actually IS (Mouse vs Joystick).
+    """
+    if length < 1:
+        return None
+
+    buf = bytearray(length)
+    try:
+        # 0x81 = Dir: IN | Type: Standard | Recipient: Interface
+        # wValue = 0x2200 (Report Descriptor)
+        device.ctrl_transfer(
+            _RECIP_INTERFACE | _REQ_TYPE_STANDARD | _DIR_IN,
+            _REQ_GET_DESCRIPTOR,
+            DESC_REPORT << 8,
+            interface_num,
+            buf,
+        )
+        return buf
+    except Exception as e:
```

Comment:
Don't catch all Exceptions. You'll get reloads and such too. You probably just want USBError.